### PR TITLE
Asset Processor: Measure, show, store, and load analysis job (CreateJob) duration metrics

### PR DESCRIPTION
**Pull request by c5h11oh**
## What does this PR do?

image
(Duration values are faked to show the sortation)

In this PR, Asset Processor measured the duration of analysis jobs (a.k.a. CreateJob), showed the accumulated analysis job duration for each source file in the Source Asset tree (sortable by duration), and store/load historical duration records to/from asset database. The work is similar to the combination of #10371 (collect and show "last processing job duration" in Jobs tab) and #10518 (save/read processing duration to/from asset database), but here it shows and persists analysis job metrics.

Changes include:
- Collect and persist analysis job duration
  - set `persistToDb` to true when calling `AssetProcessor::StatsCapture::EndCaptureStat()`
- Display analysis job duration on source asset tree in Assets tab
  - `SourceAssetTreeModel::AddOrUpdateEntry()` optionally saves the duration to `SourceAssetTreeItemData`.
  - `AssetTreeitem` passes `GetColumnCount()` and `GetDataForColumn()` requests to corresponding `AssetTreeItemData` virtual methods to do the real work.
  - Create `SourceAssetTreeItemData` that inherits `AssetTreeItemData`, overriding `GetColumnCount()` and `GetDataForColumn()`, to show the duration column.
  - `SourceAssetTreeModel::ResetModel()` has to reset the unique_ptr `m_root` so that `m_root` is a `SourceAssetTreeItemData`, and thus the above virtual methods can be called.
    - Its caller `AssetTreeModel::Reset()` resets `m_root` right before calling into it. The double reset is a waste though.
    - An alternative is to put `m_root.reset()` in `ResetModel()` of all subclasses of `AssetTreeModel` (e.g. `SourceAssetTreeModel`, `ProductAssetTreeModel`), but I think may cause errors if people want to create subclasses of `AssetTreeModel` (need to remember resetting `m_root`).
- Load historical duration records from database when launch. 
  - Done in `SourceAssetTreeModel::ResetModel()`.
- On file change and re-analyze, update the duration in the database and UI
  - Done in `SourceAssetTreeModel::AddOrUpdateEntry()`
- Sort by duration works. When sort descending, folders will be on the top and listed alphabetically
  - Create `SourceAssetTreeFilterModel` that inherits `AssetTreeFilterModel`
- Code clean up
  - Remove `SourceAssetTreeItemData::MakeShared()`. Replace the use of this method with `AZStd::make_shared<SourceAssetTreeItemData>`.
  - Unify the terms: Use "analysis job" and "processing job" to respectively refer to "CreateJob" and "ProcessJob". This aligns the global stats appeared on the top right corner of the Asset Processor.
  - Set (Source|Product)AssetTreeView's column width for better user experience.

## How was this PR tested?
Manually tested.

### CODE REVIEW

File: Code/Tools/AssetProcessor/native/ui/AssetTreeItem.cpp [hunk 1]
```diff
@@ -26,6 +27,30 @@ namespace AssetProcessor
...
+    int AssetTreeItemData::GetColumnCount() const
+    {
+        return aznumeric_cast<int>(AssetTreeColumns::Max);
+    }
+
+    QVariant AssetTreeItemData::GetDataForColumn(int column) const
+    {
+        switch (column)
+        {
+        case static_cast<int>(AssetTreeColumns::Name):
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/ui/AssetTreeItem.cpp:39]: aznumeric_cast<>() is the best practice

File: Code/Tools/AssetProcessor/native/ui/SourceAssetTreeFilterModel.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/ui/SourceAssetTreeFilterModel.cpp:9]: no needed for CPP files

File: Code/Tools/AssetProcessor/native/ui/AssetTreeItem.cpp (hunk 1 above)
Comment (c5h11oh) [reply Code/Tools/AssetProcessor/native/ui/AssetTreeItem.cpp]: fixed

File: Code/Tools/AssetProcessor/native/ui/SourceAssetTreeFilterModel.cpp (hunk 2 above)
Comment (c5h11oh) [reply Code/Tools/AssetProcessor/native/ui/SourceAssetTreeFilterModel.cpp]: removed

File: Code/Tools/AssetProcessor/native/ui/SourceAssetTreeFilterModel.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <ui/AssetTreeFilterModel.h>
+
+namespace AssetProcessor {
```
[the remark this answers is not in the document]
Comment (c5h11oh) [reply Code/Tools/AssetProcessor/native/ui/SourceAssetTreeFilterModel.h]: fixed

**Review by amzn-mike — APPROVED** [review]

**Comment by AMZN-stankowi** [comment]
We had a discussion about terminology in the Asset Processor, both the UX, C++, and documentation. We decided that we need to do a deeper dive into this later, and no action needs to be taken right now to standardize any terminology around "Create" versus "Analysis" for that step of the job process

**Pull request merged**